packages/javascript/bh-shared-ui/src/components/AnalyzeNowConfiguration/AnalyzeNowConfiguration.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright 2025 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+import { Button } from '@bloodhoundenterprise/doodleui';
+import { RequestOptions } from 'js-client-library';
+import { useState } from 'react';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import { useNotifications } from '../../providers';
+import { apiClient } from '../../utils';
+import AnalyzeNowConfirmDialog from './AnalyzeNowConfirmDialog';
+
+const requestAnalysis = (options?: RequestOptions) => {
+    return apiClient.requestAnalysis(options).then((res) => res.data);
+};
+
+const useRequestAnalysis = () => {
+    const queryClient = useQueryClient();
+    return useMutation(requestAnalysis, {
+        onSuccess: () => {
+            queryClient.invalidateQueries('datapipe-status');
+        },
+    });
+};
```

Comment:
Since you aren't using the options, no need to pass them. Also, you may as well include the method inline since it isn't reused. And you could clean up some of the code by passing in the dialog hide and including the notification hook directly, since it's only being used here. This keeps all the mutation callbacks together.

```suggestion
const useRequestAnalysis = (hideDialog: () => void) => {
  const queryClient = useQueryClient();
  const { addNotification } = useNotifications();

  return useMutation(
    () => apiClient.requestAnalysis().then((res) => res.data),
    {
      onSuccess: () => {
        queryClient.invalidateQueries("datapipe-status");
        addNotification("Analysis requested successfully.");
        hideDialog();
      },
      onError: () => {
        addNotification("There was an error requesting analysis.");
        hideDialog();
      },
    }
  );
};
```